kratos/utilities/entities_utilities.h

```diff
@@ -37,6 +39,30 @@ namespace EntitiesUtilities
      */
     void KRATOS_API(KRATOS_CORE) InitializeAllEntities(ModelPart& rModelPart);
 
+    /**
+     * @brief This method calls InitializeSolution for all the entities (conditions, elements, constraints)
+     * @param rModelPart The model part of the problem to solve
+     */
+    void KRATOS_API(KRATOS_CORE) InitializeSolutionStepAllEntities(ModelPart& rModelPart);
+
+    /**
+     * @brief This method calls FinalizeSolutionStep for all the entities (conditions, elements, constraints)
+     * @param rModelPart The model part of the problem to solve
+     */
+    void KRATOS_API(KRATOS_CORE) FinalizeSolutionStepAllEntities(ModelPart& rModelPart);
+
+    /**
+     * @brief This method calls InitializeNonLinearIteration for all the entities (conditions, elements, constraints)
+     * @param rModelPart The model part of the problem to solve
+     */
+    void KRATOS_API(KRATOS_CORE) InitializeNonLinearIterationAllEntities(ModelPart& rModelPart);
+
+    /**
+     * @brief This method calls FinalizeNonLinearIteration for all the entities (conditions, elements, constraints)
+     * @param rModelPart The model part of the problem to solve
+     */
+    void KRATOS_API(KRATOS_CORE) FinalizeNonLinearIterationAllEntities(ModelPart& rModelPart);
+
     /**
      * @brief This method initializes all the active entities
```

Comment:
```suggestion
     * @brief This method returns the appropriate TEntitytype container (elements, conditions, and nodes) from model part
```

This is not from this PR, but i just saw... This should be like this ryt?